weld/subsystem/src/main/java/org/jboss/as/weld/WeldProvider.java

```diff
@@ -87,9 +90,23 @@ public CdiImpl(Container container, BeanManagerProxy rootBeanManager, WeldDeploy
         }
 
         @Override
-        public BeanManagerProxy getBeanManager() {
+        public BeanManager getBeanManager() {
             checkContainerState(container);
             final String callerName = getCallingClassName();
+            if(callerName.startsWith("org.glassfish.soteria")) {
+                //the Java EE Security RI uses CDI.current() to perform bean lookup, however
```

Comment:
This whole seems to me as workaround. What would be proper fix for this? Make soteria not to use CDI.current()? Or is there some design problem in WildFly?

Do we have weld issue for this to make proper discussion?

Reply:
CDI.current() is not very well specified, it does not really say how been visibility sour be handled.